src/main/java/gov/cabinetofice/gapuserservice/service/OneLoginService.java

```diff
@@ -86,27 +104,93 @@ public User createNewUser(final String sub, final String email) {
         return userRepository.save(user);
     }
 
+    public String generateSecureRandomString(final Integer strLen) {
+        final String chrs = "0123456789abcdefghijklmnopqrstuvwxyz-_ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+
+        final SecureRandom secureRandom = new SecureRandom();
+
+        return secureRandom
+                .ints(strLen, 0, chrs.length())
+                .mapToObj(i -> chrs.charAt(i))
+                .collect(StringBuilder::new, StringBuilder::append, StringBuilder::append)
+                .toString();
+    }
+
     public String generateNonce() {
-         return UUID.randomUUID().toString();
+        //return Objects.equals(this.configProperties.getProfile(), "LOCAL") ? "aEwkamaos5C" : generateSecureRandomString(64);
```

Comment:
Can remove this comment

Reply:
this is actually the line we need, forgot to reenable it after doing tests